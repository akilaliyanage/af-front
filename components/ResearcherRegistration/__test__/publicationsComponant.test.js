import React  from 'react';
import ReactDOM from 'react-dom';
import {render, screen, cleanup} from '@testing-library/react'
import PublicationComponent from '../publicationsComponant'
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import expectExport from 'expect';

//clean up
afterEach(()=>cleanup())


//Test for successful rendering of the membership componant(Load without crash)
it('Chech Publication succssful rendering',() =>{
    const div = document.createElement('div')
    let yearList = [2011 , 2012 , 2013]
    ReactDOM.render(
            <BrowserRouter>
                <PublicationComponent yearList={yearList}/>
            </BrowserRouter>
    ,div)

})

//Test for successful rendering of the qualification componant - contain the select element
it('Publication Componant renders correctly', () =>{
    let yearList = [2011 , 2012 , 2013]
    render(
            <BrowserRouter>
                <PublicationComponent yearList={yearList}/>
            </BrowserRouter>
    )
    const qualificationContainer = screen.getByTestId('qualification-container')
    expectExport(qualificationContainer).toContainHTML("</select>")
})


//Compare snapshot of qualification componant for testing - any changes to the
//qualification componant will be detecteted after creating the snapshot
it('snapshot check for Publication', () =>{
    let yearList = [2011 , 2012 , 2013 , 2021 , 2019]
    const componant = renderer.create(
            <BrowserRouter>
                <PublicationComponent yearList={yearList}/>
            </BrowserRouter>
    ).toJSON();

    expect(componant).toMatchSnapshot()
    
})

