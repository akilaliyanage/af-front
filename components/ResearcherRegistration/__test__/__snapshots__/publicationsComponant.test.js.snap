// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot check for Publication 1`] = `
<div
  className="qualification-container"
  data-testid="qualification-container"
>
  <div
    className="form-row-container qualification-row"
  >
    <div
      className="form-Element-container"
    >
      <input
        className="Researcher-form-input-long"
        id="publication"
        name="publication"
        type="text"
      />
      <label>
        Publication
      </label>
    </div>
  </div>
  <div
    className="form-row-container qualification-row"
  >
    <div
      className="form-Element-container"
    >
      <input
        className="Researcher-form-input-long"
        id="source"
        name="source"
        type="text"
      />
      <label>
        Conference/Journal/Book
      </label>
    </div>
    <div
      className="form-Element-container"
    >
      <select
        className="Researcher-form-input-address-extras"
        id="pubYear"
        name="pubYear"
      >
        <option
          value={2011}
        >
          2011
        </option>
        <option
          value={2012}
        >
          2012
        </option>
        <option
          value={2013}
        >
          2013
        </option>
        <option
          value={2021}
        >
          2021
        </option>
        <option
          value={2019}
        >
          2019
        </option>
      </select>
      <label>
        Year
      </label>
    </div>
  </div>
  <div
    className="form-row-container qualification-row"
  >
    <div
      className="form-Element-container"
    >
      <input
        className="Researcher-form-input-long"
        id="pubURL"
        name="pubURL"
        type="text"
      />
      <label>
        URL
      </label>
    </div>
  </div>
</div>
`;
